### ISSUE TITLE
deadlocked threads won't die after crash from RenderingModulesSettingScreen

### ISSUE THREAD

**Issue by keturn** [MovingBlocks/Terasology#3996]
When crashing from RenderingModulesSettingScreen (crash itself to be filed in another issue), a bunch of threads don't want to go away and let the game close in peace.

Non-daemon threads include:

* Pathfinder-0
* Workboard-0
* Saving-0 — A thread calling "Saving" is maybe one we actually _do_ avoid killing? but I don't know what would be saving anything this early.
* Chunk-Generator-* 
* Chunk-Unloader-*
* Chunk-Updater-*

Additionally, this is a daemon thread, but it's waiting for something that seems like it's not coming:

```
"Thread-3" #43 daemon prio=10 os_prio=0 cpu=0.64ms elapsed=851.89s tid=0x00007f8f1a9a9800 nid=0x247c5b in Object.wait()  [0x00007f8e1a5f8000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(java.base@11.0.7/Native Method)
        - waiting on <0x00000000a174bb78> (a net.java.games.input.LinuxDeviceThread)
        at java.lang.Object.wait(java.base@11.0.7/Object.java:328)
        at net.java.games.input.LinuxDeviceThread.run(LinuxDeviceThread.java:60)
        - waiting to re-lock in wait() <0x00000000a174bb78> (a net.java.games.input.LinuxDeviceThread)
```

see also #3953 for other non-daemon-thread talk.

Where we were when crashing:

```
	at org.terasology.rendering.nui.layers.mainMenu.videoSettings.RenderingModuleSettingScreen.postInit(RenderingModuleSettingScreen.java:112)
	at org.terasology.rendering.nui.layers.mainMenu.StartPlayingScreen.lambda$initialise$3(StartPlayingScreen.java:111)
	at org.terasology.rendering.nui.ActivatableWidget.activateWidget(ActivatableWidget.java:46)
```

### Log details and game version

this is just after the DAG branch landed in develop.
